plutus-tx/src/Language/PlutusTx/Lift/Class.hs

```diff
@@ -479,3 +514,14 @@ makeLift name = do
     decl <- TH.funD 'lift clauses
     let liftDecs = [TH.InstanceD Nothing constraints (liftPir uni datatypeType) [decl]]
     pure $ typeableDecs ++ liftDecs
+
+
+-- | In case of exception, it will call `fail` in TemplateHaskell
+runTHCompile :: THCompile a -> TH.Q (a, Deps)
+runTHCompile m = do
+    res <- runExceptT $
+          flip runReaderT mempty $
+          runStateT m mempty
```

Comment:
if you're going to do the `flip` pattern to pass the reader context/state first, do it for both!
```suggestion
          flip runStateT mempty m
```